docs/modules/sql/pages/sql-reflection-configuration.adoc

```diff
@@ -0,0 +1,96 @@
+= Configuring Reflection
+:description: Clusters can be configured to restrict the capability of creating objects through reflection during SQL execution.
+
+When you have set the object format to `java`, you can configure clusters to restrict object creation using reflection during SQL execution.
+
+== Disallowing Reflection of Untrusted Java Classes
+
+To prevent Hazelcast from constructing objects of arbitrary classes through reflection, you can allow or deny reflection for Java classes based on class name, package name, or prefix using the `java-reflection-filter` parameter.
+
+When objects are constructed through reflection, the following filtering rules apply:
+
+* When the `whitelist` option is empty:
+** If the object's class name or package name is blacklisted, reflection fails.
+** Otherwise, reflection is allowed.
+* When the `whitelist` option is populated:
+** If the object's class name or package name is blacklisted, reflection fails.
+** If the object's class name or package name is whitelisted, reflection is allowed
+** Otherwise, reflection fails.
+
+When reflection fails, a `SecurityException` is thrown.
+
+By default, class names or package names that have these prefixed are allowed:
```

Comment:
there are actually 2 "defaults":
- `SqlConfig.javaReflectionFilterConfig = null` -> all classes allowed
- `SqlConfig.javaReflectionFilterConfig = new JavaSerializationFilterConfig()` -> whitelist as described here

we should make this clear somehow or maybe change the implementation, but without enabling any filtering by default because it would be a breaking change

Reply:
The Java Reflection filter operates in a manner analogous to the Java serialization filter configuration.
I don't think having a different logic is a good idea.

I ll add more info to the doc.

Reply:
no, i am wrong

Reply:
this is fine for java config but how does it map to XML and yaml?

Reply:
5532bd50a327cbb9b384c33d1cbc6332c2192147